File: app/src/main/java/vn/loitp/app/activity/animation/lottie/MenuLottieActivity.java
package vn.loitp.app.activity.animation.lottie;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.SeekBar;

import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.airbnb.lottie.LottieAnimationView;
import com.core.base.BaseFontActivity;
import com.core.utilities.LUIUtil;
import com.google.android.material.tabs.TabLayout;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import vn.loitp.app.R;

//https://www.lottiefiles.com/?page=1
public class MenuLottieActivity extends BaseFontActivity {
    private List<LottieItem> lottieItemList = new ArrayList<>();
    private ViewPager viewPager;
    private SeekBar sb;
    private SlidePagerAdapter slidePagerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewPager = findViewById(R.id.viewPager);
        sb = findViewById(R.id.sb);
        slidePagerAdapter = new SlidePagerAdapter();
        viewPager.setAdapter(slidePagerAdapter);
        TabLayout tabLayout = findViewById(R.id.tabLayout);
        tabLayout.setupWithViewPager(viewPager);
        LUIUtil.INSTANCE.changeTabsFont(tabLayout, com.core.common.Constants.INSTANCE.getFONT_PATH());
        sb.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int i, boolean b) {

            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
                viewPager.setCurrentItem(seekBar.getProgress());
            }
        });
        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int i, float v, int i1) {
            }

            @Override
            public void onPageSelected(int i) {
                sb.setProgress(i);
            }

            @Override
            public void onPageScrollStateChanged(int i) {
            }
        });
        prepareData();
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_menu_lottie;
    }

    private void prepareData() {
        getListAssetFiles("lottie");
        for (int i = 0; i < stringList.size(); i++) {
            lottieItemList.add(new LottieItem(i + " - " + stringList.get(i), "lottie/" + stringList.get(i)));
        }
        sb.setMax(lottieItemList.size());
        viewPager.getAdapter().notifyDataSetChanged();
    }

    private List<String> stringList = new ArrayList<>();

    private boolean getListAssetFiles(String path) {
        String[] list;
        try {
            list = getAssets().list(path);
            if (list.length > 0) {
                // This is a folder
                for (String file : list) {
                    if (!getListAssetFiles(path + "/" + file))
                        return false;
                    else {
                        // This is a file
                        stringList.add(file);
                    }
                }
            }
        } catch (IOException e) {
            return false;
        }
        return true;
    }

    private class SlidePagerAdapter extends PagerAdapter {
        @Override
        public Object instantiateItem(ViewGroup collection, int position) {
            LottieItem lottieItem = lottieItemList.get(position);
            LayoutInflater inflater = LayoutInflater.from(getActivity());
            ViewGroup layout = (ViewGroup) inflater.inflate(R.layout.item_lottie_view, collection, false);
            RelativeLayout rl = layout.findViewById(R.id.rl);
            LottieAnimationView lottieAnimationView = layout.findViewById(R.id.animation_view);

            lottieAnimationView.setAnimation(lottieItem.getPathAsset());
            //lottieAnimationView.useHardwareAcceleration();
            //lottieAnimationView.setScale(0.3f);

            //lottieAnimationView.playAnimation();
            lottieAnimationView.loop(true);

            rl.setOnClickListener(view -> {
                lottieAnimationView.playAnimation();
                //lottieAnimationView.loop(true);
            });

            collection.addView(layout);
            return layout;
        }

        @Override
        public void destroyItem(ViewGroup collection, int position, Object view) {
            collection.removeView((View) view);
        }

        @Override
        public int getCount() {
            return lottieItemList.size();
        }

        @Override
        public boolean isViewFromObject(View view, Object object) {
            return view == object;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return lottieItemList.get(position).getName();
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/animation/lottie/MenuLottieActivity.kt
package vn.loitp.app.activity.animation.lottie

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.airbnb.lottie.LottieAnimationView
import com.airbnb.lottie.LottieDrawable
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.common.Constants
import com.loitpcore.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_animation_lottie.*
import vn.loitp.app.R
import java.io.IOException

// https://www.lottiefiles.com/?page=1
@LogTag("MenuLottieActivity")
@IsFullScreen(false)
class MenuLottieActivity : BaseFontActivity() {
    private val lottieItemList: MutableList<LottieItem> = ArrayList()

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_animation_lottie
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
        prepareData()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.setImageResource(R.color.transparent)
            this.tvTitle?.text = MenuLottieActivity::class.java.simpleName
        }

        val slidePagerAdapter = SlidePagerAdapter()
        viewPager.adapter = slidePagerAdapter
        tabLayout.setupWithViewPager(viewPager)

        LUIUtil.changeTabsFont(tabLayout = tabLayout, fontName = Constants.FONT_PATH)

        sb.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, i: Int, b: Boolean) {}
            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {
                viewPager.currentItem = seekBar.progress
            }
        })
        viewPager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(i: Int, v: Float, i1: Int) {}
            override fun onPageSelected(i: Int) {
                sb.progress = i
            }

            override fun onPageScrollStateChanged(i: Int) {}
        })
    }

    private fun prepareData() {
        getListAssetFiles("lottie")
        for (i in stringList.indices) {
            lottieItemList.add(
                LottieItem(
                    i.toString() + " - " + stringList[i],
                    "lottie/" + stringList[i]
                )
            )
        }
        sb.max = lottieItemList.size
        viewPager.adapter?.notifyDataSetChanged()
    }

    private val stringList: MutableList<String> = ArrayList()
    private fun getListAssetFiles(path: String): Boolean {
        val list: Array<String>?
        try {
            list = assets.list(path)
            if (list?.isNotEmpty() == true) {
                // This is a folder
                for (file in list) {
                    if (!getListAssetFiles(path = "$path/$file")) return false else {
                        // This is a file
                        stringList.add(file)
                    }
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
            return false
        }
        return true
    }

    private inner class SlidePagerAdapter : PagerAdapter() {
        override fun instantiateItem(collection: ViewGroup, position: Int): Any {
            val lottieItem = lottieItemList[position]
            val inflater = LayoutInflater.from(this@MenuLottieActivity)
            val layout = inflater.inflate(R.layout.item_lottie_view, collection, false) as ViewGroup
            val rl = layout.findViewById<ViewGroup>(R.id.rl)
            val animationView: LottieAnimationView = layout.findViewById(R.id.animationView)
            animationView.setAnimation(lottieItem.pathAsset)
            // lottieAnimationView.useHardwareAcceleration();
            // lottieAnimationView.setScale(0.3f);

            // lottieAnimationView.playAnimation();
//            animationView.loop(true)
            animationView.repeatCount = LottieDrawable.INFINITE
            rl.setOnClickListener {
                animationView.playAnimation()
            }
            collection.addView(layout)
            return layout
        }

        override fun destroyItem(collection: ViewGroup, position: Int, view: Any) {
            collection.removeView(view as View)
        }

        override fun getCount(): Int {
            return lottieItemList.size
        }

        override fun isViewFromObject(view: View, `object`: Any): Boolean {
            return view === `object`
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return lottieItemList[position].name
        }
    }
}
